### ISSUE TITLE
booster DART predict

### ISSUE THREAD

**Issue by hadjipantelis** [dmlc/xgboost#3015]
In R, in the case of using dart booster, `predict(xgboostmodel, newdata)` does not use all trees by default, as hinted in the `predict.xgb.Booster` documentation. It appears that it picks at random (?) one iteration and returns the relevant results; one needs to fix `predict` such that `predict(xgboostmodel, newdata, ntreelimit = modelFit$niter)`. 

This behaviour breaks compatibility with the `tree` as well as the `glm` booster, so it is a bit baffling. Can someone please amend it or at least provided some reference about it in  the documentation?

```R
rm(list=ls())
library(xgboost) 
library(mlbench)
data(BostonHousing)

param = expand.grid( nrounds = 200, max_depth = c(6), eta = 0.006,
                     rate_drop =0.1, skip_drop = 0.1, gamma = 0, subsample = c(0.4),
                     colsample_bytree = c( 0.4), min_child_weight = c(2))
i=1
set.seed(i)
picks = sample(nrow(BostonHousing), size = 100)
dtrain <- xgboost::xgb.DMatrix(data = as.matrix(BostonHousing[-picks, -c(4,14)]),
                                                 label = BostonHousing[-picks, 14] ) 
dtest <- xgboost::xgb.DMatrix(data = as.matrix(BostonHousing[ picks, -c(4,14)]),
                                                label = BostonHousing[picks, 14] )   

watchlist <- list(train = dtrain, eval = dtest)

# browser()
out <- xgboost::xgb.train(list(max_depth = param$max_depth,
                               eta = param$eta,
                               rate_drop = param$rate_drop,
                               skip_drop = param$skip_drop,
                               min_child_weight = param$min_child_weight,
                               gamma = param$gamma,
                               subsample = param$subsample,
                               colsample_bytree = param$colsample_bytree),
                          data = dtrain,
                          nrounds = param$nrounds, objective = "reg:linear",
                          booster= "dart",
                          watchlist = watchlist)
predict(out, dtest)
qpreds = sapply(1:20, function(x) predict(out, dtest))
all( apply(qpreds,1, function(x) all(x == x[1])) ) # FALSE // Should evaluate to TRUE



sessionInfo()
#R version 3.4.3 (2017-11-30)
#Platform: x86_64-pc-linux-gnu (64-bit)
#Running under: Debian GNU/Linux 8 (jessie)
#
#Matrix products: default
#BLAS: /usr/lib/openblas-base/libblas.so.3
#LAPACK: /usr/lib/libopenblasp-r0.2.12.so
#
#locale:
# [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
# [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
# [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
# [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
# [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#
#attached base packages:
#[1] stats     graphics  grDevices utils     datasets  methods   base     
#
#other attached packages:
#[1] mlbench_2.1-1   xgboost_0.6.4.6
#
#loaded via a namespace (and not attached):
#[1] compiler_3.4.3      magrittr_1.5        Matrix_1.2-12      
#[4] tools_3.4.3         stringi_1.1.5       grid_3.4.3         
#[7] data.table_1.10.4-2 lattice_0.20-35

**Comment by hcho3** [comment]
#3556 disables dropout for `predict()` so that all trees get evaluation at testing time. Dropout is only to be used at training time.
